katsdpimager/metadata.py

```diff
@@ -0,0 +1,62 @@
+"""Create partial metadata.json for katsdptransfer"""
+
+from datetime import datetime
+from typing import Sequence, Dict
+
+import numpy as np
+import katpoint
+import katdal
+import astropy.units as u
+
+
+def target_mask(dataset: katdal.DataSet, target: katpoint.Target) -> np.ndarray:
+    """Get boolean mask of dumps where the target is being tracked."""
+    # Based on katdal.DataSet.select, but we don't actually use select because
+    # we don't want to modify the dataset.
+    scan_sensor = dataset.sensor['Observation/scan_state']
+    track_mask = (scan_sensor == 'track')
+    target_index_sensor = dataset.sensor['Observation/target_index']
+    target_index = dataset.catalogue.targets.index(target)
+    mask = (target_index_sensor == target_index) & track_mask
+    return mask
```

Comment:
For reference: see [SR-1732](https://skaafrica.atlassian.net/browse/SR-1732). I had a similar problem when trying to work out integration time for the katsdpcontim metadata.

Reply:
Yes, I remember seeing that.